app/src/main/java/com/github/se/icebreakrr/model/profile/ProfilesViewModel.kt

```diff
@@ -408,28 +410,19 @@ open class ProfilesViewModel(
     when (_pictureChangeState.value) {
       ProfilePictureState.TO_UPLOAD ->
           validateAndUploadProfilePicture(context) { url ->
-            val newProfile = _editedCurrentProfile.value!!.copy(profilePictureUrl = url)
-            updateProfile(newProfile, {}, {})
-            _selfProfile.value = newProfile
-            // todo: "_selectedProfile.value = newProfile" needs to be removed when all the other
-            // personal profile views are updated to use the selfProfile instead of the
-            // selectedProfile
-            _selectedProfile.value = newProfile
-
+            val newProfile = _editedSelfProfile.value!!.copy(profilePictureUrl = url)
+            updateProfile(newProfile, { _selfProfile.value = newProfile }, {})
             resetProfileEditionState()
           }
       TO_DELETE ->
           deleteCurrentUserProfilePicture {
-            val newProfile = _editedCurrentProfile.value?.copy(profilePictureUrl = null)
-            updateProfile(newProfile!!, {}, {})
-            _selfProfile.value = newProfile
-            _selectedProfile.value = newProfile // todo: same as above
+            val newProfile = _editedSelfProfile.value?.copy(profilePictureUrl = null)
+            updateProfile(newProfile!!, { _selfProfile.value = newProfile }, {})
             resetProfileEditionState()
           }
       else -> {
-        updateProfile(_editedCurrentProfile.value!!, {}, {})
-        _selfProfile.value = _editedCurrentProfile.value
-        _selectedProfile.value = _editedCurrentProfile.value // todo: same as above
+        updateProfile(
+            _editedSelfProfile.value!!, { _selfProfile.value = _editedSelfProfile.value }, {})
```

Comment:
Actually I thought about doing this, but instagram behavior's looks like mine : you can update your PP, but if something wrong happens during the upload, you'll actually still see your old one
So I'm not sure which implementation to follow
The idea is to show a loading screen the less possible because it's a constraint for the user

Reply:
Ok, i was maybe too concerned about the crash case to see that your implementation work as needed. Don't worry, I gave it another try and it's the right way to do it.